src/simpleppl.jl

```diff
@@ -0,0 +1,231 @@
+using AbstractPPL
+import Base.getindex
+using SparseArrays
+using Setfield
+using Setfield: PropertyLens, get
+
+"""
+    ModelState(value::NamedTuple{T}, input::NamedTuple{T}, eval::NamedTuple{T}, kind::NamedTuple{T})
+
+Record the state of the model as a struct of NamedTuples, all
+sharing the same key values, namely, those of the model parameters.
+`value` should store the initial/current value of the parameters.
+`input` stores a tuple of inputs for a given node. `eval` are the
+anonymous functions associated with each node. These might typically
+be either deterministic values or some distribution, but could an 
+arbitrary julia program. `kind` is a tuple of symbols indicating
+whether the node is a logical or stochastic node.
+"""
+struct ModelState{T}
+    value::NamedTuple{T}
+    input::NamedTuple{T}
+    eval::NamedTuple{T}
+    kind::NamedTuple{T}
+end
+
+"""
+    DAG(inputs)
+
+Struct containing the adjacency matrix for a particular model and 
+the topologically ordered vertex list.
+"""
+struct DAG
```

Comment:
Maybe consider `GraphInfo` for better consistency with `VarInfo` (i.e. model = varinfo + graphinfo):
```suggestion
struct GraphInfo
```